### PR TITLE
feature/SEAR-2209-authenticated-user-metrics

### PR DESCRIPTION
**Pull request by nbkincaid**
**Issue Link**
https://spothero.atlassian.net/browse/SEAR-2188

**Description**
In this pr, I set the initial values of authenticated_user on request context.

### CODE REVIEW

File: jose/middleware.go [hunk 1]
```diff
@@ -171,6 +172,13 @@ func validateRequiredScope(r *http.Request, params AuthParams) error {
 				return fmt.Errorf(missingRequiredScope)
 			}
 		}
+
+		// set authenticated client on request context for use in downstream metrics
+		authenticatedClient := claim.ExtractAuthenticatedClientGroup()
+		if authenticatedClient != "" {
+			newContext := context.WithValue(r.Context(), utils.AuthenticatedClientKey, authenticatedClient)
+			*r = *r.WithContext(newContext)
```
Comment (nbkincaid) [inline jose/middleware.go:180]: add test case to cover this line

File: http/metrics.go [hunk 2]
```diff
@@ -23,15 +23,9 @@ import (
 	"github.com/cep21/circuit/v3"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/spothero/tools/http/writer"
+	"github.com/spothero/tools/utils"
 )
 
-// ContextKey this type is created to avoid the linting error:
```
Comment (nbkincaid) [inline http/metrics.go:28]: This had to move because of import issues across jose and http

**Review by abhioncbr — APPROVED** [review]

**Review by theisenmark — APPROVED** [review]

**Pull request merged**